planning/exact-types/requirements.md

```
@@ -0,0 +1,1360 @@
+  built solely to extract via `infer` (e.g. `fn(...args: infer P)`,
+  `Array<infer E>`) are treated as exactness-agnostic on the matched
+  shape, so a single pattern matches both exact and inexact `T`. This
+  matches the rule already given for function patterns in §4.2.4 and
+  generalizes it to object/tuple patterns used for inference.
+
+#### 7.4.3. Distribution over unions
+
+When `T` is a union, the conditional distributes over its members and
+each distributed result has the exactness of its branch:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)